src/Workspaces/Remote/ServiceHub/ExternalAccess/UnitTesting/Api/UnitTestingRoslynServicesWrapper.cs

```diff
@@ -12,9 +13,13 @@ namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api
     {
         private readonly SolutionService _solutionService;
 
+        [Obsolete]
         public UnitTestingRoslynServicesWrapper(UnitTestingPinnedSolutionInfoWrapper pinnedSolutionInfoWrapper, UnitTestingAssetStorageWrappper assetStorageWrappper)
             => _solutionService = new SolutionService(SolutionService.CreateAssetProvider(pinnedSolutionInfoWrapper.UnderlyingObject, assetStorageWrappper.UnderlyingObject));
 
+        public UnitTestingRoslynServicesWrapper(UnitTestingPinnedSolutionInfoWrapper pinnedSolutionInfoWrapper, UnitTestingAssetStorageWrapper assetStorageWrapper)
```

Comment:
>UnitTestingAssetStorageWrapper [](start = 112, length = 30)

You shouldn't need to access asset storage. What is this method used for?

Reply:
All we want is to use the SolutionService, UnitTestingAssetStorageWrapper is used to create the solution service, and we are using SolutionService.GetSolutionAsync(PinnedSolutionService) to get Solution in Remote process.

I think the problem is, is there any elegant way to get the solution on the remote side?
How we do it now is not very graceful, 
it is done in this way, when SolutionInfo is passed [here](https://github.com/dotnet/roslyn/blob/da02b3ea3c539df05d82b362e57bbe085fc1abf7/src/VisualStudio/Core/Def/Implementation/Remote/JsonRpcConnection.cs#L108) then because PinnedSolutionInfo is an internal class, on the unit testing side (example) , we have to use the object as the input type because we don't have access to PinnedSolutionInfo. Then we have to do the deserialization manually in a [wrapper ](https://github.com/dotnet/roslyn/blob/da02b3ea3c539df05d82b362e57bbe085fc1abf7/src/Workspaces/Remote/ServiceHub/ExternalAccess/UnitTesting/Api/UnitTestingPinnedSolutionInfoWrapper.cs#L22) class,
and, in the end, use the Solution Service to restore the solution on the remote process.
This whole procedure is long and tedious, and if there is a better way then we could avoid a lot of wrapper classes and clean the code a lot

Reply:
OK I have added a UnitTestingServiceBase to handle this